pathpy/HigherOrderNetwork.py

```diff
@@ -129,21 +129,22 @@ def __init__(self, paths, k=1, separator='-', nullModel=False, method='FirstOrde
                 else:
                     # Generate names of k-order nodes v and w
                     v = p[0]
+                    w = p[1]
                     for l in range(1, k):
                         v = v + separator + p[l]
-                    w = p[1]
-                    for l in range(2, k+1):
-                        w = w + separator + p[l]
+                        w = w + separator + p[l+1]
+                """
                 if v not in self.nodes:
                     self.nodes.append(v)
                 if w not in self.nodes:
```

Comment:
Hi @benassa-de-glassa thanks for the contribution 👍 . I think you need to uncomment this part otherwise the `self.nodes` will always be empty.